build-tools/packages/build-cli/src/codeCoverageAnalysisTool/compareCodeCoverage.ts

```diff
@@ -0,0 +1,109 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import type { CoverageReport } from "./getCoverageMetrics.js";
+
+// List of packages to be ignored from code coverage analysis
+const codeCoverageComparisonIgnoreList: string[] = [
+	"experimental",
+	"examples",
+	"packages.common.core-interfaces",
+	"packages.common.core-utils",
+	"packages.common.driver-definitions",
+	"packages.common.container-definitions",
+	"packages.common.client-utils",
+];
+
+/**
+ * Type for the code coverage report generated by comparing the baseline and pr code coverage
+ */
+export interface CodeCoverageComparison {
+	/**
+	 * Name of the package
+	 */
+	packageName: string;
+	/**
+	 * Line coverage in baseline build
+	 */
+	lineCoverageInBaseline: number;
+	/**
+	 * Line coverage in pr build
+	 */
+	lineCoverageInPr: number;
+	/**
+	 * difference between line coverage in pr build and baseline build
+	 */
+	lineCoverageDiff: number;
+	/**
+	 * branch coverage in baseline build
+	 */
+	branchCoverageInBaseline: number;
+	/**
+	 * branch coverage in pr build
+	 */
+	branchCoverageInPr: number;
+	/**
+	 * difference between branch coverage in pr build and baseline build
+	 */
+	branchCoverageDiff: number;
+	/**
+	 * Flag to indicate if the package is new
+	 */
+	isNewPackage: boolean;
+}
+
+/**
+ * Compares the code coverage for pr and baseline build and returns an array of the CodeCoverageDiffReport type
+ */
+export const compareCodeCoverage = (
+	baselineCoverageReport: CoverageReport[],
+	prCoverageReport: CoverageReport[],
+): CodeCoverageComparison[] => {
+	const results: CodeCoverageComparison[] = [];
+
+	prCoverageReport.forEach((packageInPrReport: CoverageReport) => {
```

Comment:
Lint should have caught this - prefer for...of to Array.forEach.

Reply:
Fixed.